ui/src/main/webapp/src/app/shared/checkboxes.component.ts

```diff
@@ -24,168 +22,72 @@ export class CheckboxesComponent implements OnInit, OnChanges
      */
     @Input()
     valueCallback: (item: any) => string
-        = app => { throw new Error("valueCallback not yet defined." +
-                " This means Angular didn't keep the order of setting @Inputs."); };
+        = app => { throw new Error("valueCallback not yet defined.") };
 
     @Input()
     labelCallback: (item: ItemType) => string;
 
     @Input()
-    equalsCallback: (item1: ItemType, item2: ItemType) => boolean;
+    equalsCallback: (item1: ItemType, item2: ItemType) => boolean = (item1, item2) => {
+        return this.valueCallback(item1) == this.valueCallback(item2);
+    };
 
     /**
      * All available options.
      */
     @Input()
     set options(options: ItemType[]){
         this._options = options;
-        console.log("set options()", options);
+        //console.log("set options()", options);
         if (!options)
             return; // May be loaded async.
         if (!Array.isArray(options))
             throw new Error("Invalid @Input value for options: " + JSON.stringify(options));
-
-        options.forEach(
-            option => this.valueToOptionMap.set(this.valueCallback(option), option)
-        );
     }
     get options(): ItemType[]{
         return this._options;
     }
     _options: ItemType[];
 
-    // For faster lookup of what option was clicked
-    private valueToOptionMap: Map<string, ItemType> = new Map<string, ItemType>();
-
-
     /**
      * This can be either the values or a subset of options.
      */
-    @Input() //@Output()
-    checkedOptions: string[] | ItemType[] = [];
+    @Input()
+    checkedOptions: ItemType[] = [];
 
     @Output()
-    checkedOptionsChange = new EventEmitter<string[] | ItemType[]>();
+    checkedOptionsChange = new EventEmitter<ItemType[]>();
 
     shouldBeChecked(option: ItemType): boolean {
         console.log("shouldBeChecked() called.", option, this.checkedOptions);
-        let res = (() => {
-            if (!this.checkedOptions)
-                return false;
-
-            //if ((<ItemType[]>this.checkedOptions).indexOf(option) != -1)
-            if ((<ItemType[]>this.checkedOptions).some(checkedOption => this.equalsCallback(option, checkedOption)))
-                return true;
-
-            if ((<string[]>this.checkedOptions).indexOf(this.valueCallback(option)) != -1)
-                return true;
 
+        if (!this.checkedOptions)
             return false;
-        })();
-        console.log("shouldBeChecked() says " + res);
-        return res;
-    }
 
-    /**
-     * Determines whether the checked options are returned as string values only (see valueCallback),
-     * or as a subset of options (object references).
-     * This is useful when checkedOptions is empty and we don't know which of these to set.
-     */
-    @Input()
-    checkedOptionsAreValuesOnly: boolean | null = null;
+        if ((<ItemType[]>this.checkedOptions).some(checkedOption => this.equalsCallback(option, checkedOption)))
+            return true;
```

Comment:
All the string type stuff was kind of useless. Except indexOf could be a bit faster...